### PR TITLE
update & fix jsonrpc doc

### PR DESCRIPTION
**Pull request by xli**
1. add golang sdk repo link
2. fix TransactionData missing field gas_currency
3. add note to transaction type for how to create hash

### CODE REVIEW

File: json-rpc/docs/type_transaction.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ A transaction on the blockchain.
 | vm_status                 | [VMStatus](#type-vmstatus)               | The returned status of the transaction after being processed by the VM                     |
 | gas_used                  | unsigned int64 | Amount of gas used by this transaction, to know how much you paid for the transaction, you need multiply it with your RawTransaction#gas_unit_price |
 
-Note: For the gas_used, internally within the VM we scale the gas units down by 1000 in order to allow granularity of costing for instruction, but without having to use floating point numbers, but we do round-up the gas used to the nearest "1" when we convert back out.
+Note:
+* For the gas_used, internally within the VM we scale the gas units down by 1000 in order to allow granularity of costing for instruction, but without having to use floating point numbers, but we do round-up the gas used to the nearest "1" when we convert back out.
+* Formula to create hash for a signed transaction before it is executed: hex-encode(sha3-256([]byte("LIBRA::Transaction")) + []byte {0} + signed transaction bytes) ([implementation example](https://github.com/libra/libra-client-sdk-go/blob/master/libratypes/hash.go#L27))
```
Comment (davidiw) [inline json-rpc/docs/type_transaction.md:23]: what does the 0 represent?

Comment (davidiw) [reply json-rpc/docs/type_transaction.md:23]: also maybe using (byte[]) like casting is easier to understand?

**Review by davidiw — APPROVED** [review]

**Review by thefallentree — APPROVED** [review]

File: json-rpc/docs/type_transaction.md (hunk 1 above)
Comment (xli) [reply json-rpc/docs/type_transaction.md:23]: 0 represent Transaction first variant (enum) value UserTransaction

Comment (davidiw) [reply json-rpc/docs/type_transaction.md]: oh yeah, I figured, but it wasn't defined here.

also the use of []byte seems very strange in a Rust repo :P

**Pull request closed without merging**